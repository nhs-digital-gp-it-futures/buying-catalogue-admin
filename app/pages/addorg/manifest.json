{
  "title": "Find an organisation",
  "description": "Enter the correct Organisation Data Service (ODS) code to find the organisation you're looking for.",
  "isMultiQuestion": true,
  "questions": [
    {
      "id": "odsCode",
      "mainAdvice": "Enter ODS code"
    }
  ],
  "continueButtonText": "Continue"
}
