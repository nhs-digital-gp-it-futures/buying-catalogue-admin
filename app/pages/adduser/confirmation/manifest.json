{
  "backLinkText": "Back to dashboard",
  "backLinkHref": "/organisations/{{organisationId}}",
  "title": "{{userAdded}} account added",
  "description": "The user has been sent an email so they can set their password."
}
