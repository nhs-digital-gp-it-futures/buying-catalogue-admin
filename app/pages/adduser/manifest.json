{
  "title": "Add new user",
  "description": "Provide the following information to make a new user account.",
  "orgNameSubheading": "Organisation name",
  "isMultiQuestion": true,
  "questions": [
    {
      "id": "firstName",
      "mainAdvice": "First name"
    },
    {
      "id": "lastName",
      "mainAdvice": "Last name"
    },
    {
      "id": "phoneNumber",
      "mainAdvice": "Telephone number"
    },
    {
      "id": "emailAddress",
      "mainAdvice": "Email address"
    }
  ],
  "addUserButtonText": "Add user"
}
