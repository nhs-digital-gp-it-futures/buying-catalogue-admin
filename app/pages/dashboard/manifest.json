{
  "title": "Manage organisations and users",
  "description": "Edit existing account information or add new user to an organisation.",
  "columnInfo": [
    {
      "data": "Organisation name"
    },
    {
      "data": "ODS code"
    }
  ],
  "columnClass": "nhsuk-grid-column-one-half"
}
