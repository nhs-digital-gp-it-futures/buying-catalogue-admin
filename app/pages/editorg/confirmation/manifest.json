{
  "backLinkText": "Back to dashboard",
  "title": "{{organisationName}} updated",
  "description": "This organisation's details have been updated.",
  "dataTestId": "edit-organisation-confirmation"
}
