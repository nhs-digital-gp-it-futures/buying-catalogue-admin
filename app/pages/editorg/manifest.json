{
  "description": "Change the Organisation End User Agreement status for this organisation.",
  "odsCodeHeading": "ODS code",
  "organisationNameHeading": "Organisation name",
  "primaryRoleIdHeading": "Primary Role ID",
  "addressHeading": "Business address",
  "catalogueAgreementCheckboxText": "Organisation End User Agreement has been signed",
  "saveButtonText": "Save"
}
