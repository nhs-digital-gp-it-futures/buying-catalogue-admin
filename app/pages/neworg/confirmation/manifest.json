{
  "backLinkText": "Back to dashboard",
  "title": "{{orgName}} created",
  "description": "This organisation has been created on the Buying Catalogue.",
  "dataTestId": "create-org-confirmation"
}
