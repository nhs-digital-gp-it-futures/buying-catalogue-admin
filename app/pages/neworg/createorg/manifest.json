{
  "title": "Create organisation",
  "description": "Create organisation on the Buying Catalogue.",
  "organisationNameHeading": "Organisation name",
  "odsCodeHeading": "ODS code",
  "addressHeading": "Address",
  "primaryRoleIdHeading": "Primary Role ID",
  "catalogueAgreementCheckboxText": "Ensure this organisation has a signed Organisation End User Agreement before confirming",
  "saveButtonText": "Save"
}
