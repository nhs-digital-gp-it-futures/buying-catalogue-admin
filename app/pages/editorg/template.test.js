import { createTestHarness } from '../../test-utils/testHarness';
import manifest from './manifest.json';

const setup = {
  template: {
    path: 'pages/editorg/template.njk',
  },
};

const mockData = {
  organisationName: 'Greater Manchester CCG',
  odsCode: 'X01',
  primaryRoleId: 'ID123',
  address: ['12 Station Road', 'Leeds', 'West Yorkshire', 'LS15 5FG'],
  agreementSigned: false,
  saveCatalogueAgreementPostAction: '/organisations/org1',
};

const mockContext = {
  ...manifest,
  ...mockData,
  backLinkHref: '/organisations/org1',
  csrfToken: 'mockCsrfToken',
};

describe('editorg page', () => {
  it('should render a backLink to the organisation page', createTestHarness(setup, (harness) => {
    harness.request(mockContext, ($) => {
      const homepageBackLink = $('[data-test-id="go-back-link"]');
      expect(homepageBackLink.length).toEqual(1);
      expect(homepageBackLink.text().trim()).toEqual('Back');
      expect($(homepageBackLink).find('a').attr('href')).toEqual(mockContext.backLinkHref);
    });
  }));

  it('should render a title', createTestHarness(setup, (harness) => {
    harness.request(mockContext, ($) => {
      const title = $('h1[data-test-id="edit-org-page-title"]');
      expect(title.length).toEqual(1);
      expect(title.text().trim()).toEqual(`Edit ${mockContext.organisationName}`);
    });
  }));

  it('should render a description', createTestHarness(setup, (harness) => {
    harness.request(mockContext, ($) => {
      const description = $('h2[data-test-id="edit-org-page-description"]');
      expect(description.length).toEqual(1);
      expect(description.text().trim()).toEqual(mockContext.description);
    });
  }));

  it('should render organisation ods code', createTestHarness(setup, (harness) => {
    harness.request(mockContext, ($) => {
      const heading = $('h3[data-test-id="organisation-ods-code-heading"]');
      const odsCode = $('div[data-test-id="organisation-ods-code"]');
      expect(heading.length).toEqual(1);
      expect(heading.text().trim()).toEqual(mockContext.odsCodeHeading);
      expect(odsCode.length).toEqual(1);
      expect(odsCode.text().trim()).toEqual(mockContext.odsCode);
    });
  }));

  it('should render organisation name', createTestHarness(setup, (harness) => {
    harness.request(mockContext, ($) => {
      const heading = $('h3[data-test-id="organisation-name-heading"]');
      const orgName = $('div[data-test-id="organisation-name"]');
      expect(heading.length).toEqual(1);
      expect(heading.text().trim()).toEqual(mockContext.organisationNameHeading);
      expect(orgName.length).toEqual(1);
      expect(orgName.text().trim()).toEqual(mockContext.organisationName);
    });
  }));

  it('should render address', createTestHarness(setup, (harness) => {
    harness.request(mockContext, ($) => {
      const heading = $('h3[data-test-id="organisation-address-heading"]');
      const address = $('div[data-test-id="organisation-address"]');
      const addressLines = address.find('div');
      const address1 = $('[data-test-id="organisation-address-1"]');
      const address2 = $('[data-test-id="organisation-address-2"]');
      const address3 = $('[data-test-id="organisation-address-3"]');
      const address4 = $('[data-test-id="organisation-address-4"]');

      expect(heading.length).toEqual(1);
      expect(heading.text().trim()).toEqual(mockContext.addressHeading);
      expect(address.length).toEqual(1);
      expect(addressLines.length).toEqual(mockContext.address.length);
      expect(address1.length).toEqual(1);
      expect(address1.text().trim()).toEqual(mockContext.address[0]);
      expect(address2.length).toEqual(1);
      expect(address2.text().trim()).toEqual(mockContext.address[1]);
      expect(address3.length).toEqual(1);
      expect(address3.text().trim()).toEqual(mockContext.address[2]);
      expect(address4.length).toEqual(1);
      expect(address4.text().trim()).toEqual(mockContext.address[3]);
    });
  }));

  it('should render organisation primary role id', createTestHarness(setup, (harness) => {
    harness.request(mockContext, ($) => {
      const heading = $('h3[data-test-id="organisation-primary-role-id-heading"]');
      const primaryRoleID = $('div[data-test-id="organisation-primary-role-id"]');
      expect(heading.length).toEqual(1);
      expect(heading.text().trim()).toEqual(mockContext.primaryRoleIdHeading);
      expect(primaryRoleID.length).toEqual(1);
      expect(primaryRoleID.text().trim()).toEqual(mockContext.primaryRoleId);
    });
  }));

  describe('catalogue agreement signed checkbox', () => {
    it('should render a checkbox input for catalogue agreement signed', createTestHarness(setup, (harness) => {
      harness.request(mockContext, ($) => {
        const checkboxInput = $('input[data-test-id="catalogue-agreement-signed-checkbox"]');
        expect(checkboxInput.length).toEqual(1);
        expect(checkboxInput.attr('id')).toEqual('catalogue-agreement-checkbox');
        expect(checkboxInput.attr('name')).toEqual('catalogueagreementsigned');
        expect(checkboxInput.attr('type')).toEqual('checkbox');
        expect(checkboxInput.attr('checked')).toEqual(undefined);
        expect(checkboxInput.attr('value')).toEqual('false');
      });
    }));

    it('should render a checkbox label for catalogue agreement signed', createTestHarness(setup, (harness) => {
      harness.request(mockContext, ($) => {
        const checkboxLabel = $('label[data-test-id="catalogue-agreement-signed-label"]');
        expect(checkboxLabel.length).toEqual(1);
        expect(checkboxLabel.attr('for')).toEqual('catalogue-agreement-checkbox');
        expect(checkboxLabel.text().trim()).toEqual(mockContext.catalogueAgreementCheckboxText);
      });
    }));

    it('should check the checkbox if agreementSigned is true', createTestHarness(setup, (harness) => {
      const agreementSignedMockContext = { ...mockContext, agreementSigned: true };
      harness.request(agreementSignedMockContext, ($) => {
        const checkboxInput = $('input[data-test-id="catalogue-agreement-signed-checkbox"]');
        expect(checkboxInput.length).toEqual(1);
        expect(checkboxInput.attr('value')).toEqual('true');
      });
    }));
  });

  it('should render save button', createTestHarness(setup, (harness) => {
    harness.request(mockContext, ($) => {
      const saveButton = $('[data-test-id="save-button"] button');
      expect(saveButton.length).toEqual(1);
      expect(saveButton.text().trim())
        .toEqual(mockContext.saveButtonText);
    });
  }));

  it('should create form element with correct action', createTestHarness(setup, (harness) => {
    harness.request(mockContext, ($) => {
      const formElement = $('form');
      expect(formElement.length).toEqual(1);
      expect(formElement.attr('action')).toEqual(mockContext.saveCatalogueAgreementPostAction);
    });
  }));

  it('should render hidden input with csrf token', createTestHarness(setup, (harness) => {
    harness.request(mockContext, ($) => {
      const formElement = $('input[name=_csrf]');
      expect(formElement.length).toEqual(1);
      expect(formElement.attr('type')).toEqual('hidden');
      expect(formElement.attr('value')).toEqual(mockContext.csrfToken);
    });
  }));
});
