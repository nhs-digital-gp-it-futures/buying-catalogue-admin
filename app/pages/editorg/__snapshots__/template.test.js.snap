// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editorg page the page should render 1`] = `
"<main class=\\"nhsuk-main-wrapper\\" id=\\"maincontent\\" data-test-id=\\"main-content\\">
        
  <div class=\\"nhsuk-width-container\\" data-test-id=\\"edit-organisation-page\\">
    <div data-test-id=\\"go-back-link\\">
      <div class=\\"nhsuk-back-link\\">
  <a class=\\"nhsuk-back-link__link\\" href=\\"/organisations/org1\\">
  <svg class=\\"nhsuk-icon nhsuk-icon__chevron-left\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\">
    <path d=\\"M8.5 12c0-.3.1-.5.3-.7l5-5c.4-.4 1-.4 1.4 0s.4 1 0 1.4L10.9 12l4.3 4.3c.4.4.4 1 0 1.4s-1 .4-1.4 0l-5-5c-.2-.2-.3-.4-.3-.7z\\"></path>
  </svg>
  Go back</a>
</div>

    </div>
  
    <div class=\\"nhsuk-grid-row\\">
      <div class=\\"nhsuk-grid-column-two-thirds\\">
        <h1 data-test-id=\\"edit-org-page-title\\" class=\\"nhsuk-u-margin-top-7 nhsuk-u-margin-bottom-3\\">Edit Greater Manchester CCG</h1>
        <p data-test-id=\\"edit-org-page-description\\" class=\\"nhsuk-heading-l nhsuk-u-font-size-22 nhsuk-u-margin-bottom-9\\">Change the Organisation End User Agreement status for this organisation.</p>
      </div>
    </div>

    <dl class=\\"nhsuk-summary-list nhsuk-u-padding-margin-5\\">
      <div class=\\"nhsuk-summary-list__row\\">
        <dt class=\\"nhsuk-summary-list__key\\">
          Organisation name
        </dt>
        <dd class=\\"nhsuk-summary-list__value\\">
          Greater Manchester CCG
        </dd>
      </div>
      <div class=\\"nhsuk-summary-list__row\\">
        <dt class=\\"nhsuk-summary-list__key\\">
          ODS code
        </dt>
        <dd class=\\"nhsuk-summary-list__value\\">
          X01
        </dd>
      </div>
      <div class=\\"nhsuk-summary-list__row\\">
        <dt class=\\"nhsuk-summary-list__key\\">
          Primary Role ID
        </dt>
        <dd class=\\"nhsuk-summary-list__value\\">
          ID123
        </dd>
      </div>
      <div class=\\"nhsuk-summary-list__row\\">
        <dt class=\\"nhsuk-summary-list__key\\">
          Business address
        </dt>
        <dd class=\\"nhsuk-summary-list__value\\">
          
            <div data-test-id=\\"org-page-address-1\\">12 Station Road</div>
          
            <div data-test-id=\\"org-page-address-2\\">Leeds</div>
          
            <div data-test-id=\\"org-page-address-3\\">West Yorkshire</div>
          
            <div data-test-id=\\"org-page-address-4\\">LS15 5FG</div>
          
        </dd>
      </div>
    </dl>

    <form method=\\"post\\" action=\\"/organisations/org1\\">
      <input type=\\"hidden\\" name=\\"_csrf\\" value=\\"mockCsrfToken\\">
    
      <div data-test-id=\\"catalogue-agreement-checkbox\\" class=\\"nhsuk-u-margin-bottom-9\\">
        





  
<div class=\\"nhsuk-form-group\\">

  <div class=\\"nhsuk-checkboxes\\">
    
    
    
    
    
    
    <div class=\\"nhsuk-checkboxes__item\\">
      <input class=\\"nhsuk-checkboxes__input\\" id=\\"catalogue-agreement-checkbox\\" name=\\"catalogueAgreementSigned\\" type=\\"checkbox\\" value=\\"catalogueAgreementSigned\\">
      <label class=\\"nhsuk-label nhsuk-checkboxes__label\\" for=\\"catalogue-agreement-checkbox\\">
        Organisation End User Agreement has been signed
      </label>
    </div>
    
    
    
  </div>

</div>

      </div>

      <div data-test-id=\\"save-button\\">
        


  
    
  




<button class=\\"nhsuk-button nhsuk-u-margin-bottom-9\\" type=\\"submit\\">
  Save
</button>



      </div>
    </form>
  </div>

      </main>"
`;
